### PR TITLE
Add support for multi-root workspaces

### LINKED ISSUE #14: Support multi-root workspaces
Multi-root workspaces are saved as `.code-workspace` configuration files. They appear as follows in the `storage.json` file:

```json
{
  "openedPathsList": {
    "entries": [
      {
        "workspace": {
          "id": "0123456789abcdef0123456789abcdef",
          "configPath": "file:///path/to/myworkspace.code-workspace"
        }
      }
    ]
  }
}
```
(tested on VSCodium 1.62.3)

Currently, such entires are ignored. It would be nice if they could also appear in the search results.

### PR DESCRIPTION
**Pull request by vianney**
This patch adds multi-root workspaces saved as .code-workspace files to the results list.

Note: this is my first foray into rust-land, so any comment is more than welcome.

Closes #14

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -27,12 +27,20 @@ use gnome_search_provider_common::log::*;
...
+    config_path: Option<String>,
+}
+
 #[derive(Debug, Deserialize)]
 struct StorageOpenedPathsListEntry {
     #[serde(rename = "folderUri")]
     folder_uri: Option<String>,
     #[serde(rename = "fileUri")]
     file_uri: Option<String>,
+    workspace: Option<StorageOpenedPathsListEntryWorkspace>,
```
Comment (lunaryorn) [inline src/main.rs:43]: I presume it's either a folder, or a file, or a workspace?  So perhaps you could turn this into an untagged enum? See https://serde.rs/enum-representations.html#untagged

File: src/main.rs [hunk 2]
```diff
@@ -27,12 +27,20 @@ use gnome_search_provider_common::log::*;
 use gnome_search_provider_common::mainloop::*;
 use gnome_search_provider_common::matching::*;
 
+#[derive(Debug, Deserialize)]
+struct StorageOpenedPathsListEntryWorkspace {
```
Comment (lunaryorn) [inline src/main.rs:31]: I think `WorkspaceEntry` would do :slightly_smiling_face:

File: src/main.rs [hunk 3]
```diff
@@ -27,12 +27,20 @@ use gnome_search_provider_common::log::*;
...
+    id: Option<String>,
+    #[serde(rename = "configPath")]
+    config_path: Option<String>,
```
Comment (lunaryorn) [inline src/main.rs:34]: Why's this optional?  How would a workspace _without_ configuration path even work?  Does this really exist in VSCode?

File: src/main.rs [hunk 4]
```diff
@@ -74,7 +82,13 @@ impl Storage {
             let workspaces3 = paths.workspaces3.unwrap_or_default();
             entries
                 .into_iter()
-                .filter_map(|entry| entry.folder_uri)
+                .filter_map(|entry| {
+                    if let Some(workspace) = entry.workspace {
```
Comment (lunaryorn) [inline src/main.rs:86]: That's `entry.workspace.and_then(|w| w.config_path).or(entry.folder_uri)` :slightly_smiling_face:

**Review by lunaryorn — CHANGES_REQUESTED** [review]
Thanks :pray:   I've left a few comments, and I'd like to ask you to add a `CHANGELOG.md` entry for this change and the corresponding issue.

**Comment by vianney** [comment]
Thanks a lot for the review and letting me learn rust on the go. I have updated my branch with the requested changes, and updated the test suite as well.

**Review by lunaryorn — APPROVED** [review]

**Comment by lunaryorn** [comment]
@vianney Thanks a bunch :pray:

**Pull request merged**